apps/mosaico/src/Mosaico.purs

```diff
@@ -279,39 +271,30 @@ mosaicoComponent initialValues props = React.do
     case state.route of
       Routes.Frontpage -> setFrontpage (CategoryFeed frontpageCategoryLabel)
       Routes.TagPage tag -> setFrontpage (TagFeed tag)
-      Routes.SearchPage Nothing -> pure unit
       Routes.SearchPage (Just query) -> setFrontpage (SearchFeed query)
-      -- Always uses server side provided article
-      Routes.DraftPage -> setState _  { showAds = false }
-      Routes.ProfilePage -> pure unit
       Routes.ArticlePage articleId
         | Just article <- map _.article (join $ map hush $ state.article)
         , articleId == article.uuid
         -> do
           when (state.ssrPreview && _.premium article) $
             loadArticle articleId
-          setState _ { showAds = not article.removeAds && not (article.articleType == Advertorial) }
         | otherwise -> loadArticle articleId
-      Routes.MenuPage -> setState _ { showAds = false }
-      Routes.NotFoundPage _path -> setState _ { showAds = true }
       Routes.CategoryPage (Category c) -> setFrontpage (CategoryFeed c.label)
-      Routes.EpaperPage -> setState _ { showAds = true }
       Routes.StaticPage page
         | Just (StaticPageResponse r) <- state.staticPage
         , r.pageName == page
         -> when (isJust state.prevRoute) do
              foldMap (\p -> evalExternalScripts [ScriptTag $ "<script>" <> p <> "</script>"]) r.pageScript
-             setState _ { showAds = false }
         | otherwise ->
           Aff.launchAff_ do
             staticPage <- fetchStaticPage page
-            liftEffect $ setState _  { staticPage = Just staticPage, showAds = false }
+            liftEffect $ setState _  { staticPage = Just staticPage }
             case staticPage of
               StaticPageResponse r
                 | Just p <- r.pageScript -> liftEffect $ evalExternalScripts [ScriptTag $ "<script>" <> p <> "</script>"]
               _ -> mempty
-      Routes.DebugPage _ -> pure unit
       Routes.DeployPreview -> liftEffect $ setState _  { route = Routes.Frontpage }
+      _ -> pure unit
```

Comment:
I kinda prefer having explicit match lists than fallbacks in match blocks like this -- it makes it more explicit that we are not doing anything special for this route on purpose.